### ISSUE TITLE
Segfault after hard reboot (MV-SI)

### ISSUE THREAD

**Issue by bentsi** [scylladb/scylla#5129]
*Installation details*
Scylla version (or git commit hash): `3.1.0.rc8-0.20190925.d7fc7bcf9`
Cluster size: 5
OS (RHEL/CentOS/Ubuntu/AWS AMI): ami-06430761396e594ee [eu-west-1]

After a hard reboot (instance reboot using EC2 api) of the node:
```
2019-09-27 06:34:21.924 | (DisruptionEvent Severity.NORMAL): type=start name=HardRebootNode node=Node longevity-mv-si-4d-3-1-db-node-ff59e3f3-4 [18.202.241.139 \| 10.0.96.145] (seed: False)
```

Node `longevity-mv-si-4d-3-1-db-node-ff59e3f3-1 [34.251.127.159 | 10.0.21.245]` got segmentation fault
```
Sep 27 06:34:22 ip-10-0-21-245.eu-west-1.compute.internal scylla[2366]:  [shard 0] rpc - client 10.0.157.0:7000: fail to connect: Connection refused
Sep 27 06:34:23 ip-10-0-21-245.eu-west-1.compute.internal scylla[2366]:  [shard 0] storage_service - Node 10.0.96.145 state jump to normal
Sep 27 06:34:23 ip-10-0-21-245.eu-west-1.compute.internal scylla[2366]:  [shard 7] compaction - Compacting [/var/lib/scylla/data/system/peers-37f71aca7dc2383ba70672528af04d4f/mc-399-big-Data.db:level=0, /var/lib/scylla/data/system/peers-37f71aca7dc2383ba70672528af04d4f/mc-385-big-Data.db:level=0, ]
Sep 27 06:34:23 ip-10-0-21-245.eu-west-1.compute.internal scylla[2366]:  [shard 0] gossip - InetAddress 10.0.96.145 is now DOWN, status = shutdown
Sep 27 06:34:23 ip-10-0-21-245.eu-west-1.compute.internal scylla[2366]:  [shard 7] compaction - Compacted 2 sstables to [/var/lib/scylla/data/system/peers-37f71aca7dc2383ba70672528af04d4f/mc-413-big-Data.db:level=0, ]. 22175 bytes to 11227 (~50% of original) in 41ms = 0.26MB/s. ~256 total partitions merged to 1.
Sep 27 06:34:23 ip-10-0-21-245.eu-west-1.compute.internal scylla[2366]: Segmentation fault on shard 2.
Sep 27 06:34:23 ip-10-0-21-245.eu-west-1.compute.internal scylla[2366]: Backtrace:
Sep 27 06:34:23 ip-10-0-21-245.eu-west-1.compute.internal scylla[2366]: 0x00000000041871d2
Sep 27 06:34:23 ip-10-0-21-245.eu-west-1.compute.internal scylla[2366]: 0x00000000040744e5
Sep 27 06:34:23 ip-10-0-21-245.eu-west-1.compute.internal scylla[2366]: 0x00000000040747e5
Sep 27 06:34:23 ip-10-0-21-245.eu-west-1.compute.internal scylla[2366]: 0x0000000004074833
Sep 27 06:34:23 ip-10-0-21-245.eu-west-1.compute.internal scylla[2366]: 0x00007fd060e1702f
Sep 27 06:34:23 ip-10-0-21-245.eu-west-1.compute.internal scylla[2366]: 0x00000000005369bd
Sep 27 06:34:23 ip-10-0-21-245.eu-west-1.compute.internal scylla[2366]: 0x0000000001dc87b7
Sep 27 06:34:23 ip-10-0-21-245.eu-west-1.compute.internal scylla[2366]: 0x0000000002e7a39b
Sep 27 06:34:23 ip-10-0-21-245.eu-west-1.compute.internal scylla[2366]: 0x0000000000838a91
Sep 27 06:34:23 ip-10-0-21-245.eu-west-1.compute.internal kernel: reactor-2[2368]: segfault at 18 ip 00000000005369be sp 0000602016fffe90 error 4 in scylla[400000+4d70000]
Sep 27 06:34:23 ip-10-0-21-245.eu-west-1.compute.internal kernel: reactor-1[2367]: segfault at 18 ip 00000000005369be sp 000060100c69fe90 error 4 in scylla[400000+4d70000]
Sep 27 06:34:23 ip-10-0-21-245.eu-west-1.compute.internal kernel: Code: c7 03 98 52 61 04 76 09 48 8d 7b 10 e8 2b 45 ee ff 48 89 df be 50 00 00 00 e8 ce 0a af 03 48 83 c4 18 5b 5d 41 5c 41 5d c3 c3 <48> 8b 04 25 18 00 00 00 0f 0b 48 8b 04 25 18 00 00 00 0f 0b 48 8b
Sep 27 06:34:23 ip-10-0-21-245.eu-west-1.compute.internal kernel: reactor-5[2371]: segfault at 18 ip 00000000005369be sp 0000605015cbfe90 error 4 in scylla[400000+4d70000]
Sep 27 06:34:23 ip-10-0-21-245.eu-west-1.compute.internal kernel: Code: c7 03 98 52 61 04 76 09 48 8d 7b 10 e8 2b 45 ee ff 48 89 df be 50 00 00 00 e8 ce 0a af 03 48 83 c4 18 5b 5d 41 5c 41 5d c3 c3 <48> 8b 04 25 18 00 00 00 0f 0b 48 8b 04 25 18 00 00 00 0f 0b 48 8b
Sep 27 06:34:23 ip-10-0-21-245.eu-west-1.compute.internal kernel: Code: c7 03 98 52 61 04 76 09 48 8d 7b 10 e8 2b 45 ee ff 48 89 df be 50 00 00 00 e8 ce 0a af 03 48 83 c4 18 5b 5d 41 5c 41 5d c3 c3 <48> 8b 04 25 18 00 00 00 0f 0b 48 8b 04 25 18 00 00 00 0f 0b 48 8b
Sep 27 06:34:24 ip-10-0-21-245.eu-west-1.compute.internal sudo[91679]:   centos : TTY=unknown ; PWD=/home/centos ; USER=root ; COMMAND=/bin/coredumpctl --no-pager --no-legend
Sep 27 06:34:24 ip-10-0-21-245.eu-west-1.compute.internal sudo[91679]: pam_unix(sudo:session): session opened for user root by (uid=0)
Sep 27 06:34:24 ip-10-0-21-245.eu-west-1.compute.internal sudo[91679]: pam_unix(sudo:session): session closed for user root
Sep 27 06:34:40 ip-10-0-21-245.eu-west-1.compute.internal sudo[91703]:   centos : TTY=unknown ; PWD=/home/centos ; USER=root ; COMMAND=/bin/coredumpctl --no-pager --no-legend
Sep 27 06:34:40 ip-10-0-21-245.eu-west-1.compute.internal sudo[91703]: pam_unix(sudo:session): session opened for user root by (uid=0)
Sep 27 06:34:40 ip-10-0-21-245.eu-west-1.compute.internal sudo[91703]: pam_unix(sudo:session): session closed for user root
Sep 27 06:34:55 ip-10-0-21-245.eu-west-1.compute.internal sudo[91724]:   centos : TTY=unknown ; PWD=/home/centos ; USER=root ; COMMAND=/bin/coredumpctl --no-pager --no-
```
Translated backtrace:
```
void seastar::backtrace<seastar::backtrace_buffer::append_backtrace()::{lambda(seastar::frame)#1}>(seastar::backtrace_buffer::append_backtrace()::{lambda(seastar::frame)#1}&&) at /usr/include/boost/program_options/variables_map.hpp:146
seastar::backtrace_buffer::append_backtrace() at /usr/include/boost/program_options/variables_map.hpp:146
 (inlined by) print_with_backtrace at /jenkins/workspace/scylla-3.1/relocatable-pkg/scylla/seastar/build/release/../../src/core/reactor.cc:1104
seastar::print_with_backtrace(char const*) at /usr/include/boost/program_options/variables_map.hpp:146
sigsegv_action at /usr/include/boost/program_options/variables_map.hpp:146
 (inlined by) operator() at /jenkins/workspace/scylla-3.1/relocatable-pkg/scylla/seastar/build/release/../../src/core/reactor.cc:5015
 (inlined by) _FUN at /jenkins/workspace/scylla-3.1/relocatable-pkg/scylla/seastar/build/release/../../src/core/reactor.cc:5011
?? ??:0
std::default_delete<seastar::internal::do_with_state<std::vector<frozen_mutation_and_schema, std::allocator<frozen_mutation_and_schema> >, seastar::future<> > >::operator()(seastar::internal::do_with_state<std::vector<frozen_mutation_and_schema, std::allocator<frozen_mutation_and_schema> >, seastar::future<> >*) const [clone .isra.3192] at storage_proxy.cc:?
service::storage_proxy::on_down(gms::inet_address const&) at crtstuff.c:?
 (inlined by) service::storage_proxy::on_down(gms::inet_address const&) at /jenkins/workspace/scylla-3.1/relocatable-pkg/scylla/service/storage_proxy.cc:3684
operator() at /jenkins/workspace/scylla-3.1/relocatable-pkg/scylla/seastar/include/seastar/core/future.hh:768
 (inlined by) apply at /jenkins/workspace/scylla-3.1/relocatable-pkg/scylla/seastar/include/seastar/core/apply.hh:35
 (inlined by) apply<service::storage_service::notify_down(service::storage_service::inet_address)::<lambda(auto:191&&)> [with auto:191 = service::storage_service&]::<lambda()> > at /jenkins/workspace/scylla-3.1/relocatable-pkg/scylla/seastar/include/seastar/core/apply.hh:43
 (inlined by) apply_tuple<service::storage_service::notify_down(service::storage_service::inet_address)::<lambda(auto:191&&)> [with auto:191 = service::storage_service&]::<lambda()> > at /jenkins/workspace/scylla-3.1/relocatable-pkg/scylla/seastar/include/seastar/core/future.hh:1346
 (inlined by) apply<service::storage_service::notify_down(service::storage_service::inet_address)::<lambda(auto:191&&)> [with auto:191 = service::storage_service&]::<lambda()> > at /jenkins/workspace/scylla-3.1/relocatable-pkg/scylla/seastar/include/seastar/core/future.hh:1380
 (inlined by) operator() at /jenkins/workspace/scylla-3.1/relocatable-pkg/scylla/seastar/include/seastar/core/thread.hh:324
 (inlined by) call at /jenkins/workspace/scylla-3.1/relocatable-pkg/scylla/seastar/include/seastar/util/noncopyable_function.hh:71
seastar::noncopyable_function<void ()>::operator()() const at /jenkins/workspace/scylla-3.1/relocatable-pkg/scylla/seastar/build/release/../../include/seastar/util/noncopyable_function.hh:145
 (inlined by) seastar::thread_context::main() at /jenkins/workspace/scylla-3.1/relocatable-pkg/scylla/seastar/build/release/../../src/core/thread.cc:317
``` 

Test YAML: https://github.com/scylladb/scylla-cluster-tests/blob/branch-3.1/test-cases/longevity/longevity-mv-si-4days.yaml
Grafana snapshot: https://snapshot.raintank.io/dashboard/snapshot/4HmGKNMtiWSEfGNYo11cURaEyw5yBV4G

**Comment by bentsi** [comment]
All logs:
```
+-------------------------------------------------------------------------------------------------------------------------------------------------------------+
|                                           Log links for testrun with test id ff59e3f3-a13f-4d4f-a224-1d680b6764a7                                           |
+-----------------------+-------------------------------------------------------------------------------------------------------------------------------------+
| Log type              | Link                                                                                                                                |
+-----------------------+-------------------------------------------------------------------------------------------------------------------------------------+
| events                | https://cloudius-jenkins-test.s3.amazonaws.com/ff59e3f3-a13f-4d4f-a224-1d680b6764a7/events_log.zip                                  |
| job                   | https://cloudius-jenkins-test.s3.amazonaws.com/ff59e3f3-a13f-4d4f-a224-1d680b6764a7/job_log.zip                                     |
| db-cluster            | https://cloudius-jenkins-test.s3.amazonaws.com/ff59e3f3-a13f-4d4f-a224-1d680b6764a7/longevity-mv-si-4d-3-1-db-cluster-ff59e3f3.zip  |
| loader-set            | https://cloudius-jenkins-test.s3.amazonaws.com/ff59e3f3-a13f-4d4f-a224-1d680b6764a7/longevity-mv-si-4d-3-1-loader-set-ff59e3f3.zip  |
| monitor-set           | https://cloudius-jenkins-test.s3.amazonaws.com/ff59e3f3-a13f-4d4f-a224-1d680b6764a7/longevity-mv-si-4d-3-1-monitor-set-ff59e3f3.zip |
| monitoring_data_stack | https://cloudius-jenkins-test.s3.amazonaws.com/ff59e3f3-a13f-4d4f-a224-1d680b6764a7/monitoring_data_stack_branch-2.4_3.1.tar.gz     |
+-----------------------+-------------------------------------------------------------------------------------------------------------------------------------+

```

**Comment by denesb** [comment]
I'm looking at this.

**Comment by denesb** [comment]
@bentsi where is the core?

**Comment by bentsi** [comment]
there was no coredump....

**Comment by denesb** [comment]
We can't investigate without a coredump. According to the logs you posted, there was a crash so if there is no coredump it is because the system is (mis) configured to not generate one.

**Comment by bentsi** [comment]
We don't touch coredump configuration in our SCT tests.

But the reason why there was no coredump is in the database log, `systemd-coredum` itself crashed with segfault :)
```
2019-09-27T06:36:14+00:00  ip-10-0-21-245 !INFO    | kernel: systemd-coredum[91669]: segfault
```
Detailed log
```
2019-09-27T06:36:09+00:00  ip-10-0-21-245 !INFO    | scylla: [shard 0] database - Keyspace sec_index: Reading CF users id=7ce047e0-e08b-11e9-9f4a-000000000009 version=922df45a-9de4-3d13-941a-3af3cdc067e7
2019-09-27T06:36:09+00:00  ip-10-0-21-245 !INFO    | scylla: [shard 0] database - Keyspace sec_index: Reading CF users_last_name_ind_index id=7ea50250-e08b-11e9-8747-000000000009 version=ea8ee9ab-3dbe-399d-9233-58425e8c5a54
2019-09-27T06:36:09+00:00  ip-10-0-21-245 !INFO    | scylla: [shard 0] database - Keyspace sec_index: Reading CF users_address_ind_index id=f0a6b810-e0d3-11e9-b566-000000000001 version=14560a24-59cc-3a4a-a29d-9dde9a78bdd2
2019-09-27T06:36:09+00:00  ip-10-0-21-245 !INFO    | scylla: [shard 0] database - Populating Keyspace keyspace1
2019-09-27T06:36:09+00:00  ip-10-0-21-245 !INFO    | scylla: [shard 0] database - Keyspace keyspace1: Reading CF standard1 id=3c144cd0-e0d5-11e9-9515-00000000000a version=92817a9b-e95e-38b9-ad78-a219a6248154
2019-09-27T06:36:10+00:00  ip-10-0-21-245 !WARNING | scylla: [shard 5] commitlog_replayer - Corrupted file: /var/lib/scylla/commitlog/CommitLog-1-90071992547534530.log. 15519736 bytes skipped.
2019-09-27T06:36:10+00:00  ip-10-0-21-245 !WARNING | scylla: [shard 4] commitlog_replayer - Corrupted file: /var/lib/scylla/commitlog/CommitLog-1-72057594038052524.log. 29474808 bytes skipped.
2019-09-27T06:36:10+00:00  ip-10-0-21-245 !WARNING | scylla: [shard 3] commitlog_replayer - Corrupted file: /var/lib/scylla/commitlog/CommitLog-1-54043195528570621.log. 27439096 bytes skipped.
2019-09-27T06:36:10+00:00  ip-10-0-21-245 !WARNING | scylla: [shard 8] commitlog_replayer - Corrupted file: /var/lib/scylla/commitlog/CommitLog-1-144115188075980504.log. 31641592 bytes skipped.
2019-09-27T06:36:11+00:00  ip-10-0-21-245 !WARNING | scylla: [shard 1] commitlog_replayer - Corrupted file: /var/lib/scylla/commitlog/CommitLog-1-18014398509606666.log. 22192120 bytes skipped.
2019-09-27T06:36:11+00:00  ip-10-0-21-245 !NOTICE  | sudo:  centos : TTY=unknown ; PWD=/home/centos ; USER=root ; COMMAND=/bin/coredumpctl --no-pager --no-legend
2019-09-27T06:36:11+00:00  ip-10-0-21-245 !INFO    | sudo: pam_unix(sudo:session): session opened for user root by (uid=0)
2019-09-27T06:36:11+00:00  ip-10-0-21-245 !INFO    | sudo: pam_unix(sudo:session): session closed for user root
2019-09-27T06:36:11+00:00  ip-10-0-21-245 !WARNING | scylla: [shard 2] commitlog_replayer - Corrupted file: /var/lib/scylla/commitlog/CommitLog-1-36028797019088663.log. 24223736 bytes skipped.
2019-09-27T06:36:11+00:00  ip-10-0-21-245 !WARNING | scylla: [shard 7] commitlog_replayer - Corrupted file: /var/lib/scylla/commitlog/CommitLog-1-126100789566498514.log. 10219512 bytes skipped.
2019-09-27T06:36:11+00:00  ip-10-0-21-245 !WARNING | scylla: [shard 10] commitlog_replayer - Corrupted file: /var/lib/scylla/commitlog/CommitLog-1-180143985094944519.log. 18587640 bytes skipped.
2019-09-27T06:36:11+00:00  ip-10-0-21-245 !WARNING | scylla: [shard 12] commitlog_replayer - Corrupted file: /var/lib/scylla/commitlog/CommitLog-1-216172782113908420.log. 18305016 bytes skipped.
2019-09-27T06:36:13+00:00  ip-10-0-21-245 !WARNING | scylla: [shard 6] commitlog_replayer - Corrupted file: /var/lib/scylla/commitlog/CommitLog-1-108086391057016523.log. 12812280 bytes skipped.
2019-09-27T06:36:14+00:00  ip-10-0-21-245 !INFO    | kernel: systemd-coredum[91669]: segfault at 10 ip 00007fbfeccf69d1 sp 00007ffcc4a877c0 error 4 in libdw-0.176.so[7fbfeccea000+4e000]
2019-09-27T06:36:14+00:00  ip-10-0-21-245 !INFO    | kernel: Code: 48 8d 50 ff 48 83 fa fd 76 9f 49 c7 44 24 68 ff ff ff ff 48 c7 c0 ff ff ff ff e9 51 ff ff ff e8 45 cc ff ff 0f 1f 44 00 00 53 <48> 8b 47 10 48 89 fb 48 85 c0 74 13 48 83 f8 ff 74 45 8b 40 14 5b
2019-09-27T06:36:16+00:00  ip-10-0-21-245 !WARNING | scylla: [shard 13] commitlog_replayer - Corrupted file: /var/lib/scylla/commitlog/CommitLog-1-234187180623390438.log. 5177336 bytes skipped.
2019-09-27T06:36:19+00:00  ip-10-0-21-245 !WARNING | scylla: [shard 0] commitlog_replayer - Corrupted file: /var/lib/scylla/commitlog/CommitLog-1-124648.log. 28975096 bytes skipped.
2019-09-27T06:36:23+00:00  ip-10-0-21-245 !WARNING | scylla: [shard 11] commitlog_replayer - Corrupted file: /var/lib/scylla/commitlog/CommitLog-1-198158383604426509.log. 688120 bytes skipped.
2019-09-27T06:36:26+00:00  ip-10-0-21-245 !NOTICE  | sudo:  centos : TTY=unknown ; PWD=/home/centos ; USER=root ; COMMAND=/bin/coredumpctl --no-pager --no-legend
2019-09-27T06:36:26+00:00  ip-10-0-21-245 !INFO    | sudo: pam_unix(sudo:session): session opened for user root by (uid=0)
2019-09-27T06:36:26+00:00  ip-10-0-21-245 !INFO    | sudo: pam_unix(sudo:session): session closed for user root
2019-09-27T06:36:41+00:00  ip-10-0-21-245 !NOTICE  | sudo:  centos : TTY=unknown ; PWD=/home/centos ; USER=root ; COMMAND=/bin/coredumpctl --no-pager --no-legend
2019-09-27T06:36:41+00:00  ip-10-0-21-245 !INFO    | sudo: pam_unix(sudo:session): session opened for user root by (uid=0)
2019-09-27T06:36:41+00:00  ip-10-0-21-245 !INFO    | sudo: pam_unix(sudo:session): session closed for user root
2019-09-27T06:36:46+00:00  ip-10-0-21-245 !WARNING | scylla: [shard 9] commitlog_replayer - Corrupted file: /var/lib/scylla/commitlog/CommitLog-1-162129586585462631.log. 19984376 bytes skipped.
2019-09-27T06:36:52+00:00  ip-10-0-21-245 !INFO    | scylla: [shard 0] commitlog_replayer - Log replay complete, 22517385 replayed mutations (0 invalid, 0 skipped)
2019-09-27T06:36:55+00:00  ip-10-0-21-245 !INFO    | scylla: [shard 1] compaction - Compacting [/var/lib/scylla/data/sec_index/users_initials_ind_index-80a82551e08b11e9b46e000000000005/mc-1597-big-Data.db:level=0, /var/lib/scylla/data/sec_index/users_initials_ind_index-80a82551e08b11e9b46e000000000005/mc-1583-big-Data.db:level=0, ]
2019-09-27T06:36:55+00:00  ip-10-0-21-245 !INFO    | scylla: [shard 4] compaction - Compacting [/var/lib/scylla/data/mview/users_by_email-6a370f20e08b11e98ea4000000000009/mc-1656-big-Data.db:level=0, /var/lib/scylla/data/mview/users_by_email-6a370f20e08b11e98ea4000000000009/mc-1642-big-Data.db:level=0, ]
```
And there were corruptions in the commit log...

**Comment by denesb** [comment]
> We don't touch coredump configuration in our SCT tests.
>
> But the reason why there was no coredump is in the database log, systemd-coredum itself crashed with segfault :)

:grin: 

Anyway we need a core to determine why the crash happened, so please try to reproduce.

**Comment by slivne** [comment]
closing this issue till we have a reproducer

**Comment by bhalevy** [comment]
This may be a duplicate of #5032 
0d31c6da624f0f3594dec4e22f8312308d6da5a2 is not included in 3.1.0.rc8-0.20190925.d7fc7bcf9
